cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractResourceReporter.java

```diff
@@ -76,16 +89,15 @@ protected final ScheduledExecutorService executor() {
     return executor;
   }
 
-  protected void sendMetrics(String context, int containers, int memory, int vcores, String runId) {
-    LOG.trace("Reporting resources in context {}: (containers, memory, vcores) = ({}, {}, {})",
-              context, containers, memory, vcores);
-    MetricsCollector collector = collectionService.getCollector(MetricsScope.SYSTEM, context, runId);
-    collector.gauge(METRIC_CONTAINERS, containers);
-    collector.gauge(METRIC_MEMORY_USAGE, memory);
-    collector.gauge(METRIC_VIRTUAL_CORE_USAGE, vcores);
+  protected void sendMetrics(Map<String, String> context, int containers, int memory, int vcores) {
+    LOG.trace("Reporting resources: (containers, memory, vcores) = ({}, {}, {})", containers, memory, vcores);
+    MetricsCollector metricsCollector = programMetricsCollectors.getUnchecked(context);
```

Comment:
So there is no System metrics scope anymore? So can users not have a metrics name METRICS_CONTAINERS et al?